core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -499,6 +499,12 @@ class SnappyParser(session: SnappySession)
   }
 
   protected final def relationFactor: Rule1[LogicalPlan] = rule {
+    identifier ~ ws ~ '(' ~ ws ~ (expression * commaSep) ~ ws ~ ')' ~>
+        ((fnname: String, e: Any) =>
+          e match {
+            case ve: Vector[Expression] => UnresolvedTableValuedFunction(fnname.toLowerCase(), ve)
+            case _ => UnresolvedTableValuedFunction(fnname, Seq.empty)
+          }) |
```

Comment:
Also the "ws" placements are incorrect. Convention is ws following a token and all non-terminals (so all non-terminals already have a following ws): `identifier ~ '(' ~ ws ~ (expression * commaSep) ~ ')' ~ ws ~>`